### PR TITLE
docs(components): [select] use new display tag

### PR DESCRIPTION
**Pull request by wzc520pyfm**
Related: #10754 

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### <samp>🤖 Generated by Copilot at e8caf41</samp>

This pull request updates the documentation and comments of the Select component and its subcomponents (`Option` and `Option Group`) in `element-plus`. It adds type information, descriptions and consistency for the attributes, events, slots and methods of the components, and removes some deprecated or unused attributes. The purpose of this pull request is to improve the readability and usability of the documentation and code for developers and users.

## Related Issue

Fixes #\_\_\_.

### <samp>🤖 Generated by Copilot at e8caf41</samp>

*  Update the documentation of the Select component in `docs/en-US/component/select.md` to use a more consistent and informative format and add type information ([link](https://github.com/element-plus/element-plus/pull/13301/files?diff=unified&w=0#diff-1fe792429b649feda5d215381b79aa885bdc2ccba1f3dceb0ae4a454ff58dba5L122-R228))
*  Add description comments for the props of the Option Group, Option and Select components in `packages/components/select/src` ([link](https://github.com/element-plus/element-plus/pull/13301/files?diff=unified&w=0#diff-33c46490231f28214b88a69e3e74603b07037f8c644ab8be8d4ed3eebda97489L34-R41), [link](https://github.com/element-plus/element-plus/pull/13301/files?diff=unified&w=0#diff-c785b7a55fd001ef0209b2db5278f1e0f9705491acc4dd8a73778361e38ab49fL40-R55), [link](https://github.com/element-plus/element-plus/pull/13301/files?diff=unified&w=0#diff-9bfd95a8f0051f829d273a71e6ae89d2f5b583a6bdb2d808dd5b2e22709b2540L347-R549))

### CODE REVIEW

File: docs/en-US/component/select.md [hunk 1]
```diff
@@ -119,104 +119,110 @@ select/value-key
...
+| persistent                          | when select dropdown is inactive and `persistent` is `false`, select dropdown will be destroyed                             | ^[boolean]                                                                                                                                                                  | true             |
+| automatic-dropdown                  | for non-filterable Select, this prop decides if the option menu pops up when the input is focused                           | ^[boolean]                                                                                                                                                                  | false            |
+| clear-icon                          | Custom clear icon component                                                                                                 | ^[string] / ^[object]`Component`                                                                                                                                            | CircleClose      |
```
Comment (btea) [inline docs/en-US/component/select.md:157]: `Custom clear icon component`  Perhaps the first letter is not capitalized, consistent with other descriptions.  And the comments in the code below are the same.

Comment (wzc520pyfm) [reply docs/en-US/component/select.md]: updated

File: docs/en-US/component/select.md [hunk 2]
```diff
@@ -119,104 +119,110 @@ select/value-key
...
 | -------------- | ------------------------------------------------------------- | ----------------------------------------- |
-| change         | triggers when the selected value changes                      | current selected value                    |
-| visible-change | triggers when the dropdown appears/disappears                 | true when it appears, and false otherwise |
-| remove-tag     | triggers when a tag is removed in multiple mode               | removed tag value                         |
-| clear          | triggers when the clear icon is clicked in a clearable Select | —                                         |
-| blur           | triggers when Input blurs                                     | (event: FocusEvent)                       |
-| focus          | triggers when Input focuses                                   | (event: FocusEvent)                       |
+| change         | triggers when the selected value changes                      | ^[Function]`(value: any) => void`         |
+| visible-change | triggers when the dropdown appears/disappears                 | ^[Function]`(visible: Boolean) => void`   |
```
Comment (btea) [inline docs/en-US/component/select.md:178]:
```suggestion
| visible-change | triggers when the dropdown appears/disappears                 | ^[Function]`(visible: boolean) => void`   |
```

File: docs/en-US/component/select.md [hunk 3]
```diff
@@ -119,104 +119,110 @@ select/value-key
...
+## Select API
+
+### Select Attributes
+
+| Name                                | Description                                                                                                                 | Type                                                                                                                                                                        | Default          |
+| ----------------------------------- | --------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------- |
+| model-value / v-model               | binding value                                                                                                               | ^[string] / ^[number] / ^[boolean] / ^[object]`Array \| Object`                                                                                                             | —                |
```
Comment (btea) [inline docs/en-US/component/select.md:128]:
```suggestion
| model-value / v-model               | binding value                                                                                                               | ^[string] / ^[number] / ^[boolean] / ^[object] / ^[array]`                                                                                                             | —                |
```

File: docs/en-US/component/select.md (hunk 2 above)
Comment (wzc520pyfm) [reply docs/en-US/component/select.md]: updated

File: docs/en-US/component/select.md (hunk 3 above)
Comment (wzc520pyfm) [reply docs/en-US/component/select.md]: updated

File: packages/components/select/src/option.vue [hunk 4]
```diff
@@ -37,16 +37,22 @@ export default defineComponent({
...
+     */
     value: {
       required: true,
       type: [String, Number, Boolean, Object],
     },
+    /**
+     * @description label of option, same as `value` if omitted
+     */
     label: [String, Number],
     created: Boolean,
```
Comment (btea) [inline packages/components/select/src/option.vue:51]: This prop doesn't seem to work. 🤔

Comment (wzc520pyfm) [reply packages/components/select/src/option.vue:51]: It works in the following files：`select.vue` `useOption.ts` `useSelect.ts`.
It works to [Element Plus Example](https://element-plus.org/en-US/component/select.html#create-new-items), But this example has bug(see: https://github.com/element-plus/element-plus/pull/12635). If not set `created`, the select component will not automatically focus on the newly created option.

**Review by btea — APPROVED** [review]

**Review by ryuhangyeong — APPROVED** [review]

**Pull request merged**